types/node.lua

```diff
@@ -271,50 +269,88 @@ function nodetypes.nnode:outputYourself (typesetter, line)
    end
 end
 
-function nodetypes.nnode:toText ()
+--- Returns the text content of the nnode.
+-- Contrary to the parent class, this is the actual text content of the node,
+-- not a text representation of the node.
+--
+-- @treturn string The text content of the nnode.
+function nnode:toText ()
    return self.text
 end
 
-nodetypes.unshaped = pl.class(nodetypes.nnode)
-nodetypes.unshaped.type = "unshaped"
+--- An unshaped node is a text node that has not been shaped yet.
+--
+-- Derived from `nnode`.
+--
+-- Properties is_unshaped is true.
+-- Note that is_nnode is NOT true, as an unshaped node is not a representable node yet.
+--
+-- @type unshaped
+
+local unshaped = pl.class(nnode)
+unshaped.type = "unshaped"
 
-function nodetypes.unshaped:_init (spec)
+--- Constructor
+--
+-- @tparam table spec A table with the properties of the unshaped node.
+
+function unshaped:_init (spec)
    self:super(spec)
    self.width = nil
 end
 
-function nodetypes.unshaped:__tostring ()
+function unshaped:__tostring ()
    return "U(" .. self:toText() .. ")"
 end
 
-getmetatable(nodetypes.unshaped).__index = function (_, _)
+getmetatable(unshaped).__index = function (_, _)
    -- if k == "width" then SU.error("Can't get width of unshaped node", true) end
    -- TODO: No idea why porting to proper Penlight classes this ^^^^^^ started
    -- killing everything. Perhaps because this function started working and would
    -- actually need to return rawget(self, k) or something?
 end
 
-function nodetypes.unshaped:shape ()
+--- Shapes the text of the unshaped node.
+-- This is done by calling the current shaper with the text and options of the node.
+-- The result is a list of nnodes inheriting the parent of the unshaped node.
+-- The notion of parent is used by the hyphenation logic and discretionaries.
+--
+-- @treturn table A list of nnodes representing the shaped text.
+function unshaped:shape ()
```

Comment:
In relation to that [discussion](https://github.com/sile-typesetter/sile/issues/1965#issuecomment-2782731163)

Unless mistaken, that the only method (that is, besides `SILE.outputter` in output routines, a topic worth discussing separately) when we invoke an external (global) object currently from a node, the `SILE.shaper`. The later will eventually need the language.